src/pages/index.js

```diff
@@ -5,123 +5,260 @@ import PopupWithForm from "../components/PopupWithForm.js";
+    userInfo.setAvatar(userData.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
 // PopupWithForm
 // PopupWithForm
 
+const newCardPopup = new PopupWithForm({
+  popupSelector: "#add-card-modal",
+  handleFormSubmit: handleAddCardFormSubmit,
+});
+
 const editProfilePopup = new PopupWithForm({
   popupSelector: "#edit-modal",
   handleFormSubmit: handleProfileEditSubmit,
 });
-editProfilePopup.setEventListeners();
 
-const newCardPopup = new PopupWithForm({
-  popupSelector: "#add-card-modal",
-  handleFormSubmit: handleAddCardFormSubmit,
+const avatarPopup = new PopupWithForm({
+  popupSelector: "#profile-avatar-modal",
+  handleFormSubmit: handleAvatarFormSubmit,
 });
-newCardPopup.setEventListeners();
 
 // PopupWithImage
 // PopupWithImage
 
 const newImagePopup = new PopupWithImage("#image-preview-modal");
-newImagePopup.setEventListeners();
 
-// UserInfo
-// UserInfo
+// DeletePopup
+// DeletePopup
 
-const userInfo = new UserInfo(profileNameElement, jobElement);
+const cardDeletePopup = new DeleteConfirm("#delete-card", deleteCardForm);
 
-// Validation
-// Validation
+// Event Listeners
+// Event Listeners
 
-const editFormValidator = new FormValidator(settings, profileEditModal);
-const addFormValidator = new FormValidator(settings, addCardModal);
+profileFormElement.addEventListener("submit", (event) => {
+  event.preventDefault();
+});
+newImagePopup.setEventListeners();
+cardDeletePopup.setEventListeners();
+editProfilePopup.setEventListeners();
+newCardPopup.setEventListeners();
+avatarPopup.setEventListeners();
+
+// Form Validation
+// Form Validation
+
+const editFormValidator = new FormValidator(settings, profileFormElement);
 editFormValidator.enableValidation();
+
+const addFormValidator = new FormValidator(settings, addCardForm);
 addFormValidator.enableValidation();
 
-//Functions
-//Functions
+const avatarFormValidator = new FormValidator(settings, editAvatarForm);
+avatarFormValidator.enableValidation();
 
-function handleProfileEditSubmit(userData) {
-  userInfo.setUserInfo(userData);
-  editProfilePopup.close();
+// Modals
+// Modals
+
+function handleImageClick(name, link) {
+  newImagePopup.open({ name, link });
 }
 
-function prefillProfileData() {
-  const { name, description } = userInfo.getUserInfo();
-  nameInput.value = name;
-  jobInput.value = description;
-  editFormValidator.resetValidation();
+addNewCardButton.addEventListener("click", () => {
+  newCardPopup.open();
+});
+
+editAvatarButton.addEventListener("click", () => {
+  avatarPopup.open();
+});
+
+confirmDeleteBtn.addEventListener("click", () => {
+  cardDeletePopup.close();
+});
+
+profileEditBtn.addEventListener("click", (e) => {
   editProfilePopup.open();
+  e.preventDefault();
+  const newUserInfo = userInfo.getUserInfo();
+  document.querySelector("#profile-name-input").value = newUserInfo.name;
+  document.querySelector("#profile-description-input").value =
+    newUserInfo.description;
```

Comment:
**Could be improved**

It’s better to find **constants** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine